server/src/main/java/org/apache/druid/metadata/SqlSegmentsMetadataQuery.java

```diff
@@ -438,6 +466,11 @@ private UnmodifiableIterator<DataSegment> retrieveSegmentsInIntervalsBatch(
       appendConditionForIntervalsAndMatchMode(sb, intervals, matchMode, connector);
     }
 
+    // Add the used_status_last_updated time filter only for unused segments when maxUsedFlagLastUpdatedTime is non-null.
+    if (!used && maxUsedFlagLastUpdatedTime != null) {
```

Comment:
nit: could assign this expression  to local variable `boolean filterByUsedStatusLastUpdated = !used && maxUsedFlagLastUpdatedTime != null`, so we dont have to compute again. No biggie obviously though.